integration/benchmark/cmd/ci/main.go

```diff
@@ -163,60 +167,95 @@ func main() {
 	if err != nil {
 		log.Fatal().Err(err).Msgf("unable to init loader")
 	}
+
+	// run load
 	lg.Start()
 
-	testStartTime := time.Now()
-	time.Sleep(loadCase.duration)
-	lg.Stop()
-	testEndTime := time.Now()
+	testRunDuration := time.NewTimer(loadCase.duration)
+	defer testRunDuration.Stop()
+
+	// prepare data slices
+	dataSlices := make(chan dataSlice)
```

Comment:
you need to buffer this channel, otherwise `recordTransactionData` will stuck on first send.